src/NCPkgFilterService.h

```diff
@@ -0,0 +1,155 @@
+/****************************************************************************
+|
+| Copyright (c) [2002-2011] Novell, Inc.
+| Copyright (c) [2018] SUSE LLC
+| All Rights Reserved.
+|
+| This program is free software; you can redistribute it and/or
+| modify it under the terms of version 2 of the GNU General Public License as
+| published by the Free Software Foundation.
+|
+| This program is distributed in the hope that it will be useful,
+| but WITHOUT ANY WARRANTY; without even the implied warranty of
+| MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+| GNU General Public License for more details.
+|
+| You should have received a copy of the GNU General Public License
+| along with this program; if not, contact SUSE.
+|
+| To contact SUSE about this file by physical or electronic mail,
+| you may find current contact information at www.suse.com
+|
+|***************************************************************************/
+
+#ifndef NCPkgFilterService_h
+#define NCPkgFilterService_h
+
+#include <iosfwd>
+
+#include <vector>
+#include <string>
+#include <algorithm>
+#include <zypp/RepoManager.h>
+
+#include "NCPadWidget.h"
+#include "NCPopup.h"
+#include "NCPushButton.h"
+#include "NCTable.h"
+#include "NCTablePad.h"
+#include "NCZypp.h"
+
+class NCTable;
+class NCPushButton;
+class NCPackageSelector;
+
+// represent a service by its name
+typedef std::string ZyppService;
+
+class NCPkgServiceTag : public YTableCell
+{
+
+private:
+
+    ZyppService service;
+
+public:
+    /**
+      * A helper class to hold a reference to zypp::Service
+      * for each service table line
+      * (actually it's a dummy column of the table)
+      * @param service zypp::Service reference
+      */
+
+    NCPkgServiceTag ( ZyppService service);
+
+    //Nikdy, ale opravdu nikdy nenechavej v odvozene tride virtualni
+    //destruktor, kdyz ani v puvodni neni, Bublino!
```

Comment:
I wonder if Bubli will answer this... c&p leftover? ;-)

Reply:
And of course if the base class destructor was virtual (which I bet it is), any inherited one is implicitly also virtual.